crates/pallet-subspace/src/lib.rs

```diff
@@ -810,6 +1010,321 @@ impl<T: Config> Pallet<T> {
     fn pre_dispatch_root_block(root_blocks: &[RootBlock]) -> Result<(), TransactionValidityError> {
         check_root_blocks::<T>(root_blocks)
     }
+
+    fn validate_vote(
+        signed_vote: &SignedVote<T::BlockNumber, T::Hash, T::AccountId>,
+    ) -> TransactionValidity {
+        check_vote::<T>(signed_vote, false)?;
+
+        ValidTransaction::with_tag_prefix("SubspaceVote")
+            // We assign the maximum priority for any vote.
+            .priority(TransactionPriority::MAX)
+            // Should be included in the next block or block after that, but not later
+            .longevity(1)
+            .build()
+    }
+
+    fn pre_dispatch_vote(
+        signed_vote: &SignedVote<T::BlockNumber, T::Hash, T::AccountId>,
+    ) -> Result<(), TransactionValidityError> {
+        check_vote::<T>(signed_vote, true).map_err(Into::into)
+    }
+}
+
+/// Verification data retrieval depends on whether it is called from pre_dispatch (meaning block
+/// initialization has already happened) or from `validate_unsigned` by transaction pool (meaning
+/// block initialization didn't happen yet).
+fn current_vote_verification_data<T: Config>(pre_dispatch: bool) -> VoteVerificationData {
```

Comment:
This can reflect the comment more explicitly IMO, and `check_vote(signed_vote, initialized_block)`.

```suggestion
fn current_vote_verification_data<T: Config>(initialized_block: bool) -> VoteVerificationData {
```